contracts/governance/treasury/TreasuryPool.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.6;
+
+
+import {Utils} from "@kyber.network/utils-sc/contracts/Utils.sol";
+import {PermissionAdmin} from '@kyber.network/utils-sc/contracts/PermissionAdmin.sol';
+import {IERC20Ext} from "@kyber.network/utils-sc/contracts/IERC20Ext.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {ITreasuryPool} from "../../interfaces/ITreasuryPool.sol";
+
+contract TreasuryPool is ITreasuryPool, PermissionAdmin, Utils {
+  using SafeERC20 for IERC20Ext;
+
+  mapping(address => bool) private _isAuthorizedStrategy;
+  address[] private _authorizedStrategies;
+
+  constructor(address _admin, address[] memory _strategies) PermissionAdmin(_admin) {
+    for(uint256 i = 0; i < _strategies.length; i++) {
+      _authorizedStrategy(_strategies[i]);
+    }
+  }
+
+  receive() external payable {}
+
+  function authorizeStrategies(address[] calldata _strategies) external override onlyAdmin {
+    for(uint256 i = 0; i < _strategies.length; i++) {
+      _authorizedStrategy(_strategies[i]);
+    }
+  }
+
+  function unauthorizeStrategies(address[] calldata _strategies) external override onlyAdmin {
+    for(uint256 i = 0; i < _strategies.length; i++) {
+      _unauthorizedStrategy(_strategies[i]);
+    }
+  }
+
+  function replaceStrategy(address _oldStrategy, address _newStrategy) external override onlyAdmin {
+    _unauthorizedStrategy(_oldStrategy);
+    _authorizedStrategy(_newStrategy);
+  }
+
+  function withdrawFunds(
+    IERC20Ext[] calldata _tokens,
+    uint256[] calldata _amounts,
+    address payable _recipient
+  ) external override onlyAdmin {
```

Comment:
do we allow authorized strategies to call this function? If yes, then need to
- remove `onlyAdmin` and change to
- `require(msg.sender == admin || _isAuthorizedStrategy[msg.sender], "not authorized");`

Reply:
ah yeah, should be only authorized strategies